library/asn1write.c

```diff
@@ -82,8 +82,10 @@ int mbedtls_asn1_write_len( unsigned char **p, unsigned char *start, size_t len
         *--(*p) = 0x83;
         return( 4 );
     }
-
+	
+#if ((UINT_MAX) != 0xffffffffu)//relevant only for 64 bit platforms. on 32 bit platforms, size_t should always be 32 bits
```

Comment:
Could you please avoid using inline comments with `//`? Instead you can use a block comment before the line of code with:
```
/*
 *
 */
```